src/util/config.js

```diff
@@ -28,26 +28,34 @@ export default function getConfig(overrideConfig) {
 	config.API_SERVER_ROOT_URL = `${config.API_PROTOCOL}://${config.API_HOST}`;
 
 	// currently all config is available syncronously, so resolve immediately
-	return Promise.resolve(overrideConfig || config)
+	return Promise.resolve({ ...config, ...overrideConfig })
 		.then(validateConfig);
 }
 
 function validateConfig(config) {
-	if (!config) {
-		console.log(chalk.red('No config loaded'));
-		return false;
-	}
-	if (!config.oauth || !config.oauth.secret || !config.oauth.key) {
-		console.log(chalk.red(`MUPWEB_OAUTH_SECRET and MUPWEB_OAUTH_KEY must be set as environment variables
-- get the values from an admin in #web-platform on Slack`));
-		return false;
-	}
-	if (!config.PHOTO_SCALER_SALT) {
-		console.log(chalk.red(`PHOTO_SCALER_SALT must be set as an environment variable
-- get the value from an admin in #web-platform on Slack`));
-		return false;
+	console.log(config.API_PROTOCOL, 'protocol');
+	const oauthError = new Error('get oauth secrets from web platform team');
+	const configSchema = Joi.object().keys({
+		DEV_SERVER_PORT: Joi.number().integer().max(65535),
```

Comment:
looks like there's a few that are missing, but I can't tell if they're intentionally omitted and if so why:
 
`DEV_SERVER_HOST`,
`DEVBOX_IP`,
`NODE_ENV`,
`COVERALLS_REPO_TOKEN`,
`COVERALLS_SERVICE_NAME`,
`ASSET_SERVER_HOST`,
`ASSET_SERVER_PORT`,
`MUPWEB_OAUTH_KEY`,
`MUPWEB_OAUTH_SECRET`